src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -193,11 +194,21 @@
                 this.uploadFilesToAdr = outputFileTypes
                     .filter(key => this.uploadFiles.hasOwnProperty(key))
             },
+            stopPolling(id) {
+                clearInterval(id)
+            },
             async handleDownloadResult(downloadResults) {
-              if (this.open && this.downloadIsReady()) {
-                  await this.getUploadMetadata(downloadResults.downloadId)
-                  this.sendUploadFilesToADR();
-              }
+                if (this.open) {
+                    if (this.downloadIsReady()) {
+                        await this.getUploadMetadata(downloadResults.downloadId)
+                        this.stopPolling(downloadResults.statusPollId)
+                        this.sendUploadFilesToADR();
+                    }
+
+                    if (downloadResults.error && downloadResults.statusPollId > -1) {
+                        this.stopPolling(downloadResults.statusPollId)
+                    }
```

Comment:
One question on the conditionals here: if there a case where we get to this block and we _don't_ want to stop polling?

Reply:
Yes, downloadIsReady is always false even when one out of two download resources is complete, which means we will never get to stop polling the completed resource.